pkg/kp/pcstore/consul_store_test.go

```diff
@@ -913,10 +913,125 @@ func TestClosedChangeChannelResultsInTermination(t *testing.T) {
 	}
 }
 
+type RecordingSyncer struct {
+	InitialClusters []fields.ID
+
+	SyncClusterCalls   []fields.ID
+	DeleteClusterCalls []fields.ID
+
+	// Used to signal when SyncCluster is called so tests can have non-racey
+	// timeouts
+	SyncSignal chan<- struct{}
+}
+
+func (r *RecordingSyncer) GetInitialClusters() ([]fields.ID, error) {
+	return r.InitialClusters, nil
+}
+
+func (r *RecordingSyncer) SyncCluster(pc *fields.PodCluster, labeledPods []labels.Labeled) error {
+	r.SyncClusterCalls = append(r.SyncClusterCalls, pc.ID)
+	if r.SyncSignal != nil {
+		r.SyncSignal <- struct{}{}
+	}
+	return nil
+}
+
+func (r *RecordingSyncer) DeleteCluster(id fields.ID) error {
+	r.DeleteClusterCalls = append(r.DeleteClusterCalls, id)
+	return nil
+}
+
+func (r *RecordingSyncer) Type() ConcreteSyncerType {
+	return "recording"
+}
+
+func TestWatchAndSync(t *testing.T) {
+	store := consulStoreWithFakeKV()
+	quit := make(chan struct{})
+	syncSignal := make(chan struct{})
+	defer close(quit)
+
+	example := examplePodCluster()
+	pc1, err := store.Create(
+		example.PodID,
+		example.AvailabilityZone,
+		"name1",
+		example.PodSelector,
+		example.Annotations,
+		kptest.NewSession(),
+	)
+	if err != nil {
+		t.Fatalf("Couldn't create test pod cluster: %s", err)
+	}
+
+	pc2, err := store.Create(
+		example.PodID,
+		example.AvailabilityZone,
+		"name2",
+		example.PodSelector,
+		example.Annotations,
+		kptest.NewSession(),
+	)
+	if err != nil {
+		t.Fatalf("Couldn't create test pod cluster: %s", err)
+	}
+
+	// Include one of the 2 clusters as initial
+	syncer := &RecordingSyncer{
+		InitialClusters: []fields.ID{pc1.ID},
+		SyncSignal:      syncSignal,
+	}
+
+	go func() {
+		err := store.WatchAndSync(syncer, quit)
+		if err != nil {
+			t.Fatalf("Couldn't start WatchAndSync(): %s", err)
+		}
+	}()
+
+	expectedSyncCount := 2
+	actualSyncCount := 0
+	for actualSyncCount < expectedSyncCount {
+		select {
+		case <-time.After(1 * time.Second):
+			t.Fatalf("Timed out waiting for sync to happen, there were %d syncs", actualSyncCount)
```

Comment:
it may be useful if we are debugging a test failure to show the contents of SyncClusterCalls, if that's not too hard?